Initial documentation (#46)
aerogear/graphql-testx: README.md

# graphql-testx

<p align="center">
  <img width="128" src="./logo/graphql-testx.png">
</p>

> graphql-testx is a full-featured GraphQL server, with the minimum configuration required, based on Graphback and Apollo Server, used for testing GraphQL client applications or libraries. It should not be confused with mocking alternatives because graphql-testx offers persistent data between queries and mutation using in-memory SQLite database.

Reviewer: ```suggestion
graphql-testx is a full-featured GraphQL server, based on Graphback and Apollo Server. With the minimum configuration required, you have a server ready for testing GraphQL client applications or libraries.
Unlike mocking alternatives, graphql-testx offers persistent data between queries and mutation using in-memory SQLite database.
```

![graphql-testx](https://user-images.githubusercontent.com/7964685/69070551-9dc31980-0a28-11ea-8b55-97707b26693c.png)

> 

Reviewer: ```suggestion

Read our [contributing guide](CONTRIBUTING.md) if you're looking to contribute.

```

## Getting Started

### Install graphql-testx

> Attention!: not yet released

Using npm:

```
npm install graphql-testx
```

or yarn:

```
yarn add graphql-testx
```

### Create and Start the server

> The only thing you need to pass to `TestxServer` is the Data Model because `TestxServer` is using Graphback under the hood, we are using the same [Data Model Definition](https://graphback.dev/docs/datamodel) as Graphback.

Reviewer: ```suggestion
Create a testx server is quite simple, you just need to instantiate a `TestxServer` passing your data model. Note that `TestxServer` is using Graphback under the hood (to automagically generate the resolvers and type definitions), so take care to use the same [Data Model Definition](https://graphback.dev/docs/datamodel) as Graphback.

See the example:
```

Reviewer 2: We can even document testx in graphback

Author: @vinifarias not 100% sure about this correction, could we maybe write it like this?
```suggestion
To create a testx server is quite simple, you just need to instantiate a `TestxServer` object passing your data model. Note that `TestxServer` is using Graphback under the hood to automatically generate the resolvers and type definitions, so it's using the same [Data Model Definition](https://graphback.dev/docs/datamodel) as Graphback.
```

Author: @wtrocki yes I think documenting testx in graphback is a good thing, the only problem is that we will have to keep the documentation updated in two places

Reviewer 3: @b1zzu I think the code speaks for itself in this example, I would nearly start the section with the code example and then have a small paragraph underneath. I kinda find sentences in dev docs like `to do x is quite simple, simply pass x to y ...` are kinda redundant and time wasting. They are more suited to blog posts. <Opinion> In docs you wanna be more straight to the point.

'Create a `TestxServer` instance, passing in your Type Definitions/Data Model. Under the hood, Graphback is used to generate a full GraphQL Schema and resolvers. See the [Graphback Docs on Data Model Definition](https://graphback.dev/docs/datamodel)'

Reviewer 2: @b1zzu No duplication in this case. Testx will need to document all framework stuff. 
In graphback we will create testing chapter where we will cover how we test the integration and performance of the solution using testx without replicating content.

Graphback will also need serparate generator that will generate testx tests for the backend

Author: @darahayes @vinifarias using suggestions, I've moved the description under the example to give more visibility to the example, and I've shorted the description, wdyt?

Author: @wtrocki so is more how we could use testx to test graphback right? The only issue I see is that testx is bounded to a specific version of graphback which will make hard to use it to test graphback

Author: I'm going to merge it because some documentation is always better than nothing, but I keep this conversation open so that we can still improve it
